qiita_plugins/target_gene/tgp/util.py

```diff
@@ -145,9 +114,8 @@ def complete_job(server_url, job_id, payload):
     global JOB_COMPLETED
     JOB_COMPLETED = True
     # Create the URL where we have to post the results
-    url = "%s/qiita_db/jobs/%s/complete/" % (server_url, job_id)
     json_payload = dumps(payload)
-    execute_request_retry(requests.post, url, data=json_payload)
+    qclient.post("/qiita_db/jobs/%s/complete/" % job_id, data=json_payload)
```

Comment:
Same as above - these are three examples, but there are a lot more calls that it will make the entire code just messy - and the API should not be extremely dynamic, otherwise it looses its power..